### PR TITLE
Fix data race for hooks

### PR DESCRIPTION
**Pull request by ChimeraCoder**
This adds a test for the bug identified in https://github.com/sirupsen/logrus/issues/611, and a corresponding fix.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 logrus
+*.swp
+*.swo
+*.swn
```
Comment (dmathieu) [inline .gitignore]: Could you add that to your own global gitignore file and not here?

File: logger.go [hunk 2]
```diff
@@ -315,3 +315,9 @@ func (logger *Logger) level() Level {
 func (logger *Logger) SetLevel(level Level) {
 	atomic.StoreUint32((*uint32)(&logger.Level), uint32(level))
 }
+
+func (logger *Logger) AddHook(hook Hook) {
```
Comment (dmathieu) [inline logger.go]: This is only used by the test. It shouldn't be a public method. Does it even need to exist at all?

File: .gitignore (hunk 1 above)
Comment (ChimeraCoder) [inline .gitignore]: (removed)

**Comment by ChimeraCoder** [comment]
Removed the `.gitignore` update from this PR; this should be good to go now!

**Comment by ChimeraCoder** [comment]
Thanks!

**Pull request merged**